authentication/tests/test_authentication.py

```diff
@@ -1,23 +1,107 @@
-from django.test import TestCase
+from backend.tests_utils import BaseTestCase
 
 from authentication.models import User
 
 
-class AuthenticationTestCase(TestCase):
+class AuthenticationTestCase(BaseTestCase):
     """Test the authentication logic."""
 
-    # TODO: finish this lol
-
-    def setUp(self):
-        User.objects.create_user(
-            "15veaux",
-            "Florian",
-            "Veaux",
-            "florian.veaux@mines-paristech.fr",
-            "password",
-            "1996-08-28",
-            2015,
+    fixtures = ("test_authentication.yaml",)
+
+    # These tokens were generated with the private key corresponding to `authentication.keys.public.PUBLIC_KEY`. This
+    # private key should be in the `sso` project.
+    # One can use the script in the `sso_server` project to generate those tokens, and https://jwt.io/ to debug them.
+    # The audience of these tokens is `portail` (it was of course edited in INVALID_AUDIENCE_TOKEN).
+    # Except for `VALID_17UNKNOWN_TOKEN`, the user of these tokens is `17simple`.
+
+    # tokens generated with `sso_server.tests.generate_fake_tokens("portail", "17simple")` in the `sso` project.
+    tokens = {
+        "INVALID_SIGNATURE_TOKEN": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOjQ3MjYwMjE5ODQuNzk3MzMzLCJpc3MiOiJzc29fc2VydmVyIiwiYXVkIjoicG9ydGFpbCIsImp0aSI6ImJjOTM4Yzc1NjE4NTRiZTU4MTc4YzZiNzVmNWZjNWNkIiwidXNlciI6IjE3c2ltcGxlIn0.Xu4m9AA3R_y_7RN_7v3rrNq9NQ6f0aeOCTbvQp-HZBR6PhQamhSyAj9vZD05oTLl-ManzDvIXjg8QbKZh3-N0HoO5yWUHtM0CKWJUQ07NptH_VldOtvbc8Fb4Lhzr03ScxtQslaUc6DXraWd3GurDCUGRzw9NT7_VybycofwucjvQjE1gFbjqN7TnxAdb8Tye5w-89ge-1FGZvsqgZyvupScdEe7B15QUe2UTm-RsDDV-RG91B6TBoTduFpTzp3oqzrkwONurKjK929Z-Ndq6AHOP17Cj_SL4R6rxlpVsQ50j0LMODhxwBfIHnQsARiLdwGcThl6LgZLYLfciAAAAA",
```

Comment:
Maybe use a function to encode the payload here so we know what's in the tokens

Reply:
Well, actually, it is more or less well explained above.

These test tokens could be generated in the `Portail`, but this requires to give the project a pair of public / private key, etc.
For now, it works well like this. I will do it but in another PR before production.

Reply:
Please at least add in comment the unencoded tokens